openevsehttp/__main__.py

```diff
@@ -876,19 +876,21 @@ async def set_led_brightness(self, level: int) -> None:
     async def set_divert_mode(self, mode: str = "fast") -> None:
         """Set the divert mode."""
         url = f"{self.url}divertmode"
-
         if mode not in ["fast", "eco"]:
-            _LOGGER.error("Invalid value for charge_mode: %s", mode)
+            _LOGGER.error("Invalid value for divert mode: %s", mode)
             raise ValueError
-
         _LOGGER.debug("Setting divert mode to %s", mode)
-
         # convert text to int
         new_mode = divert_mode[mode]
-        data = "divertmode=" + str(new_mode)
+
+        data = f"divertmode={new_mode}"
+        
+        # Use text/plain content type for the request
+        headers = {"Content-Type": "text/plain"}
+        
         response = await self.process_request(
-            url=url, method="post", data=data
-        )  # noqa: E501
+            url=url, method="post", data=data, headers=headers
```

Comment:
```suggestion
            url=url, method="post", rapi=data
```

the `data` param sends via the `json` parameter in aiohttp, the `rapi` parameter in `process_request` should send plain text. No headers _should_ be needed.